dashboard/app/controllers/openai_chat_controller.rb

```diff
@@ -39,7 +49,40 @@ def validated_level?
     params[:type].present? && params[:type] == 'validation'
   end
 
-  def get_validated_level_test_file_contents(level_id)
+  def add_content_to_system_prompt(system_prompt, level_instructions, test_file_contents)
+    if level_instructions.present?
+      system_prompt += "\n Here are the student instructions for this level: #{level_instructions}"
+    end
+
+    if test_file_contents.present?
+      system_prompt += "\n The contents of the test file are: #{test_file_contents}"
+    end
+
+    system_prompt
+  end
+
+  private def prepend_system_prompt(system_prompt, messages)
+    system_prompt_message = {
+      content: system_prompt,
+      role: "system"
+    }
+
+    messages.unshift(system_prompt_message)
+    messages
+  end
+
+  private def read_file_from_s3(key_path)
+    if [:development, :test].include?(rack_env)
```

Comment:
is this to make local testing easier? Why do we need this for `test` too?

Reply:
Locally, this is so I can try out an updated system prompt without replacing the contents in production. (AI TA might have needed this in test for their test suite, but we don't.) I'm removing the :test env here for now.

Reply:
nice catch and good call! the `:test` part looks like a mistake on our end because it would have no effect in automated testing and would potentially break running our tests locally.